projects/cdk/directives/auto-focus/handlers/ios.handler.ts

```diff
@@ -37,42 +42,78 @@ export class TuiIosAutofocusHandler extends AbstractTuiAutofocusHandler {
     }
 
     private iosWebkitAutofocus(): void {
-        const fakeInput: HTMLElement = this.renderer.createElement('input');
-
-        fakeInput.style.position = 'absolute';
-        fakeInput.style.opacity = '0';
-        fakeInput.style.height = '0';
+        const fakeInput: HTMLInputElement = this.makeFakeInput();
+        const duration = this.getDurationTimeBeforeFocus();
 
-        const blurHandler = (): void => fakeInput.focus();
+        const blurHandler = (): void => fakeInput.focus({preventScroll: true});
         const focusHandler = (): void => {
-            setTimeout(() => {
-                this.element.focus();
-
-                /**
-                 * @note:
-                 * We can't remove the element immediately, because it breaks flow
-                 */
-                setTimeout(() => {
-                    fakeInput.removeEventListener('blur', blurHandler);
-                    fakeInput.removeEventListener('focus', focusHandler);
+            clearTimeout(this.fakeFocusTimeoutId);
+
+            this.fakeFocusTimeoutId = this.windowRef.setTimeout(() => {
+                clearTimeout(this.elementFocusTimeoutId);
+
+                fakeInput.removeEventListener('blur', blurHandler);
+                fakeInput.removeEventListener('focus', focusHandler);
+
+                this.elementFocusTimeoutId = this.windowRef.setTimeout(() => {
+                    this.element.focus({preventScroll: false});
                     fakeInput.remove();
-                });
+                }, duration);
             });
         };
 
-        /**
-         * @note: ping-pong eager strategy hack
-         * After creating an element and bringing it into DOM,
-         * the browser automatically focuses on the invisible element.
-         * And then, after focus is triggered, we try to focus on target element, and if we managed to refocus,
-         * then we try to focus again on an invisible element, so that the keyboard slowly appears.
-         * This ping pong allows the keyboard to not overlap the modal window.
-         */
         fakeInput.addEventListener('blur', blurHandler, {once: true});
         fakeInput.addEventListener('focus', focusHandler);
 
-        this.element.parentElement?.appendChild(fakeInput);
+        if (this.insideDialog()) {
```

Comment:
Is this branching really important? Maybe just always append it to body?

Reply:
fixed